src/juju/jimm/api.ts

```diff
@@ -72,3 +80,40 @@ export const checkRelations = async (
   }
   return await conn.facades.jimM.checkRelations(tuples);
 };
+
+export function migrateModel(
+  conn: ConnectionWithFacades,
+  modelName: string,
+  targetController: string,
+) {
+  return new Promise<InitiateMigrationResults>((resolve, reject) => {
+    if (conn?.facades?.jimM) {
+      conn.facades.jimM
+        .migrateModel(modelName, targetController)
+        .then((migrateModelResponse) => resolve(migrateModelResponse))
+        .catch((error) => reject(error));
+    } else {
+      reject(new Error(Label.NO_JIMM));
+    }
+  });
+}
+
```

Comment:
These methods can be significantly simplified by just making them `async`. Same goes for `listMigrationTargets`.

```suggestion
export async function migrateModel(
  conn: ConnectionWithFacades,
  modelName: string,
  targetController: string,
) {
  if (!conn?.facades?.jimM) {
    throw new Error(Label.NO_JIMM);
  }
  
  return conn.facades.jimM
    .migrateModel(modelName, targetController);
}

```